includes/class-sensei-course-enrolment-manager.php

```diff
@@ -0,0 +1,148 @@
+class Sensei_Course_Enrolment_Manager {
+	const COURSE_ENROLMENT_SITE_SALT_OPTION = 'sensei_course_enrolment_site_salt';
+
+	/**
+	 * All course enrolment providers.
+	 *
+	 * @var Sensei_Course_Enrolment_Provider_Interface[]
+	 */
+	private static $enrolment_providers;
+
+	/**
+	 * Gets the descriptive name of the provider by ID.
+	 *
+	 * @param string $provider_id Unique identifier of the enrolment provider.
+	 *
+	 * @return string|false
+	 */
+	public static function get_enrolment_provider_name_by_id( $provider_id ) {
+		$provider = self::get_enrolment_provider_by_id( $provider_id );
+		if ( ! $provider ) {
+			return false;
+		}
+
+		$provider_class = get_class( $provider );
+
+		return $provider_class::get_name();
+	}
+
+	/**
+	 * Gets the enrolment provider object by its ID.
+	 *
+	 * @param string $provider_id Unique identifier of the enrolment provider.
+	 *
+	 * @return Sensei_Course_Enrolment_Provider_Interface|false
+	 */
+	public static function get_enrolment_provider_by_id( $provider_id ) {
+		$all_providers = self::get_all_enrolment_providers();
+		if ( ! isset( $all_providers[ $provider_id ] ) ) {
+			return false;
+		}
+
+		return $all_providers[ $provider_id ];
+	}
+
+	/**
+	 * Check if we should use the legacy enrolment check. Legacy check uses course
+	 * progress to determine enrolment.
+	 *
+	 * @return bool
+	 */
+	public static function use_legacy_enrolment_check() {
+		$use_legacy_enrolment_check = false;
+
+		// Check if WCPC is around but not offering enrolment providers (an old version).
+		if (
+			class_exists( '\Sensei_WC_Paid_Courses\Sensei_WC_Paid_Courses' ) &&
+			! class_exists( '\Sensei_WC_Paid_Courses\Course_Enrolment_Providers' )
+		) {
+			$use_legacy_enrolment_check = true;
+		}
+
+		return apply_filters( 'sensei_legacy_enrolment_check', $use_legacy_enrolment_check );
+	}
+
+	/**
+	 * Get an array of all the instantiated course enrolment providers.
+	 *
+	 * @return Sensei_Course_Enrolment_Provider_Interface[]
+	 */
+	public static function get_all_enrolment_providers() {
+		if ( ! isset( self::$enrolment_providers ) ) {
+			self::$enrolment_providers = [];
+
+			/**
+			 * Fetch all registered course enrolment providers.
+			 *
+			 * @param string[] $provider_classes List of enrolment providers classes.
+			 *
+			 * @since 3.0.0
+			 */
+			$provider_classes = apply_filters( 'sensei_course_enrolment_providers', [] );
+			foreach ( $provider_classes as $provider_class ) {
+				if ( ! class_exists( $provider_class ) || ! is_a( $provider_class, 'Sensei_Course_Enrolment_Provider_Interface', true ) ) {
+					continue;
+				}
+
+				self::$enrolment_providers[ $provider_class::get_id() ] = new $provider_class();
+			}
+		}
+
+		return self::$enrolment_providers;
+	}
+
+	/**
+	 * Gets the site course enrolment salt that can be used to invalidate all enrolments.
+	 *
+	 * @return string
+	 */
+	public static function get_site_salt() {
+		$enrolment_salt = \get_option( self::COURSE_ENROLMENT_SITE_SALT_OPTION );
```

Comment:
Probably it's more a question mine about PHP than a problem. :P
But, why we need the `\` on the get_option and not on the apply_filters in the method above (both global functions and called inner a static method)?

Reply:
This isn't necessary 😃. I was used to working in a non-root namespace in WCPC. 

Doing it in non-namespaced files doesn't hurt, but definitely isn't needed. In namespaced files there is a fallback to the global namespace (that I've found sometimes doesn't work), but generally best practice to avoid ambiguity.

Reply:
Hum! Thank you for your explanation =)